src/trajectory/util.cpp

```diff
@@ -83,7 +92,51 @@ bool checkStateSpace(const statespace::StateSpace* _stateSpace)
   }
 }
 
-} // (anonymous) namespace
+void checkValidityOfSpaceAndTrajectory(
```

Comment:
Under `control/ros` we currently support only trajectories/robots with R1 and SO2 joints. We could potentially support other types of joints but that does not have to be part of this PR.
 https://github.com/personalrobotics/aikido/blob/67ab2db106391a65dcb959787e42e01811bb5257/include/aikido/control/ros/Conversions.hpp#L44

Reply:
I'm okay with keeping higher DOF joints out of this, but this function still seems really similar to `checkStateSpace` aside from some better error messages. Is the main difference that `checkStateSpace` permits higher DOF joints?

I still don't think the `space` parameter is necessary, since you can take it from the trajectory. I think this function really only looks at the trajectory's state space, so it needs a better name if we decide that it needs to be kept.